genie-agent/src/main/java/com/netflix/genie/agent/configs/ValidationConfig.java

```diff
@@ -15,7 +15,7 @@
  *     limitations under the License.
  *
  */
-package com.netflix.genie.agent;
+package com.netflix.genie.agent.configs;
```

Comment:
I was keeping each config in the package where the components are.
I like it because I don't have to make implementation public just to make them visible from the config in a different package (`such as `....agent.configs`).
Because this config creates a component that is "global" to the whole agent, I placed it in `...agent`.

I'm not in favor of having a `config` package that only contains a subset of the configs.
And if we move all configs here, there's stuff we may be forced to to make `public` as a consequence.

Reply:
The flip side and why I don't favor moving the configs around is it makes it hard to find out where everything in the app is coming from. Even today I had to search for where various beans were defined which could easily lead to me redefining them. Spring already breaks the `public/private/protected` nature of the config files anyway by using annotations/reflection. Generally speaking in community jars I've looked at this is how the spring configs are packaged. Especially the boot auto configure configs they're all sitting in one package within a given jar.

Reply:
So should we remove `@Component`, `@Service` from all the beans instantiated that way and put *every* bean into a config in this config package?

Reply:
You've hit on the conundrum I keep going back and forth on with this stuff. Originally with Genie 3 development I took a hard line and said we should put all `bean` definitions in config files to remove and dependency on frameworks from our class definitions leaving them as POJO's. You can see remnants of this with `ServicesConfig` in `genie-web`.

This is fine and well till you start trying to use more features of frameworks beyond just DI. For example you start using `@Transactional` or `@Validated` or `@EventListener`. Well now for these annotations to have any impact they need to be beans why have the bean definitions in a config file for separation of concerns when it's obvious you're relying on the framework anyway?

This is a weird topic that is bleeding through into this PR so I'm going to remove moving this file for now. We need to come to a convention for how we want to do things within our project. My initial opinion is to do something like:

If only an internal class just mark it with the proper `Stereotype` along with any conditionals if we envision it being overridden by someone. For anything we're overriding from the framework itself (validation, resource handlers, etc) that aren't just our classes we should put them in `Config` classes in a central locations. We can discuss though